### PR TITLE
a11y, ux: Change input[type=number] to inputmode

### LINKED ISSUE #14309: Don't use `<input type="number">`
I'm fairly certain we're using `<input type="number">` in Cockpit. We shouldn't. It has a ton of UX (including scrolling changing the number and unintended rounding) and a11y issues.

Instead, we should use `<input type="text" inputmode="numeric" pattern="[0-9]*">`

The awesome GOV.UK Design site has a lot of relevant information on the subject:
https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

### PR DESCRIPTION
**Pull request by garrett**
Fixes #14309

Details:
https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

### CODE REVIEW

**Comment by garrett** [comment]
I'm keeping input[type=number] in the CSS in case someone uses it in the future. (We should try to avoid doing so, but it would be worse if it would be unusable in some browsers.)

**Comment by martinpitt** [comment]
Doesn't build:
```
/home/user/cockpit-project/bots/make-checkout-workdir/pkg/machines/components/memorySelectRow.jsx
  66:37  error  Unknown property 'inputmode' found, use 'inputMode' instead  react/no-unknown-property
```

**Comment by garrett** [comment]
> Doesn't build

Whoops. Sorry. Not enough coffee this morning. Squash pushed with a working version. :wink: :coffee:

File: node_modules/README [hunk 1]
```diff
@@ -1,9 +0,0 @@
-# Cockpit Javascript Dependencies
```
Comment (martinpitt) [inline node_modules/README:1]: Argh, this slipped in again.. Can you please FP again to restore the file?

Comment (garrett) [reply node_modules/README:1]: Oof. That awful file. Sorry. It feels like I'm constantly fighting it every single time I make any changes.

**Pull request merged**